src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -264,23 +266,26 @@ private unsafe void SetAccessOrWriteTime(string path, DateTimeOffset time, bool
             const long TicksPerSecond = TicksPerMillisecond * 1000;
             long nanoseconds = (time.UtcDateTime.Ticks - DateTimeOffset.UnixEpoch.Ticks - seconds * TicksPerSecond) * NanosecondsPerTick;
 
-            if (isAccessTime)
-            {
-                buf[0].TvSec = seconds;
-                buf[0].TvNsec = nanoseconds;
-                buf[1].TvSec = _fileStatus.MTime;
-                buf[1].TvNsec = _fileStatus.MTimeNsec;
-            }
-            else
+            buf[0].TvSec = seconds;
+            buf[0].TvNsec = nanoseconds;
+            buf[1].TvSec = seconds;
+            buf[1].TvNsec = nanoseconds;
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))
```

Comment:
Typically modifying source files for a specific platform involves creating a separate `<file>.<platform>.cs` instead. In this case it would be `FileStatus.Browser.cs` to put browser specific logic.

Reply:
Hm, I guess I could also do this `#if TARGET_BROWSER` https://github.com/dotnet/runtime/blob/master/src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs#L47